src/common/iceberg.cpp

```diff
@@ -251,6 +264,39 @@ string IcebergSnapshot::GetTableVersion(const string &meta_path, FileSystem &fs,
 	}
 }
 
+string IcebergSnapshot::GuessTableVersion(const string &meta_path, FileSystem &fs, string &table_version, string &metadata_compression_codec, string &version_format = DEFAULT_TABLE_VERSION_FORMAT) {
+	string selected_metadata;
+	string version_pattern = "*"; // TODO: Different "table_version" strings could customize this
+	string compression_suffix = "";
+	if (metadata_compression_codec == "gzip") {
+		compression_suffix = ".gz";
+	}
+	
+	for(auto try_format : StringUtil::Split(version_format, ',')) {
+		auto glob_pattern = StringUtil::Format(try_format, version_pattern, compression_suffix);
+		
+		auto found_versions = fs.Glob(fs.JoinPath(meta_path, glob_pattern));
+		if(found_versions.size() > 0) {
+			selected_metadata = PickTableVersion(found_versions, version_pattern, glob_pattern);
+			if(!selected_metadata.empty()) {  // Found one
+				return selected_metadata;
+			}
+		}
+	}
+	
+	throw IOException(
+	        "Could not guess Iceberg table version using '%s' compression and format(s): '%s'",
+	        metadata_compression_codec, version_format);
+}
+
+string IcebergSnapshot::PickTableVersion(vector<string> &found_metadata, string &version_pattern, string &glob) {
+	// TODO: Different "table_version" strings could customize this
+	// For now: just sort the versions and take the largest
+	std::sort(found_metadata.begin(), found_metadata.end());
+	return found_metadata.back();
```

Comment:
<https://en.cppreference.com/w/cpp/container/vector/back>
> Calling back on an empty container causes [undefined behavior](https://en.cppreference.com/w/cpp/language/ub).

We should probably check if `found_metadata` is empty and return `string()` explicitly instead

Reply:
Thanks! I had moved that up to the caller and probably shouldn't have.